File: LeetCode75-sols/Two_Pointers/CPP/Container_With_Most_Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left_ptr = 0;
        int right_ptr = height.size() - 1;
        int max_area = 0;
        while (left_ptr < right_ptr) {
            if ((right_ptr - left_ptr) * (min(height[left_ptr], height[right_ptr])) > max_area) {
                max_area = (right_ptr - left_ptr) * (min(height[left_ptr], height[right_ptr]));
            }
            if (height[left_ptr] < height[right_ptr]) {
                left_ptr++;
            }
            else {
                right_ptr--;
            }
        }
        return max_area;
    }
};

File: LeetCode75-sols/Two_Pointers/Python/Container_With_Most_Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left_ptr = 0
        right_ptr = len(height) - 1
        max_area = 0
        while (left_ptr < right_ptr):
            curr_wdw_area = (right_ptr - left_ptr) * (min(height[left_ptr],height[right_ptr]))
            if (curr_wdw_area > max_area):
                max_area = curr_wdw_area
            if (height[left_ptr] < height[right_ptr]):
                left_ptr += 1
            else:
                right_ptr -= 1
        return max_area
